raiden/smart_contracts/TokenNetwork.sol

```diff
@@ -0,0 +1,697 @@
+
+        // The lock must not have expired, it does not matter how far in the future it would
+        // have expired. We compare the expiration block with the block at which
+        // the secret has been registered on chain.
+        require(expiration_block > secret_registry.getSecretBlockHeight(secret));
+        require(hashlock == keccak256(secret));
+
+        lockhash = keccak256(expiration_block, locked_amount, hashlock);
+        computed_locksroot = computeMerkleRoot(lockhash, merkle_proof);
+
+        // Note that unlocked locks have to be re-unlocked after a `transferUpdate` with a
+        // locksroot that does not contain this lock.
+        require(partner_state.locksroot == computed_locksroot);
+
+        // A lock can be unlocked only once per participant and per balance proof. In case there
+        // is another updateTransfer that has occured after the locks have been initially
+        // unlocked, we have to unlock those locks again. This is why the mapping key contains
+        // the nonce (to account for the updateTransfer balance proof) and the lock hashlock.
+        // Note that we cannot clear the storage for the locks that have been previously
+        // unlocked but have been invalidated by a new balance proof.
+        key = keccak256(partner_state.nonce, hashlock);
+        require(!partner_state.unlocked_locks[key]);
+        partner_state.unlocked_locks[key] = true;
+
+        // Finally change the amount of owed tokens
+        // This implementation allows for each transfer to be set only once, so
+        // it's safe to update the transferred_amount in place.
+        partner_state.transferred_amount += locked_amount;
+
+        ChannelUnlocked(channel_identifier, partner, partner_state.transferred_amount);
+    }
+
+    /// @notice Settles the balance between the two parties
+    function settleChannel(
+        uint256 channel_identifier,
+        address participant1,
+        address participant2)
+        isClosed(channel_identifier)
+        timeoutOver(channel_identifier)
+        public
+    {
+        uint256 participant1_amount;
+        uint256 participant2_amount;
+        uint256 total_deposit;
+
+        Participant memory participant1_state = channels[channel_identifier].participants[participant1];
+        Participant memory participant2_state = channels[channel_identifier].participants[participant2];
+
+        // Make sure the addresses are channel participant addresses
+        require(participant1_state.initialized);
+        require(participant2_state.initialized);
+
+        // Direct token transfers done through the token `transfer` function
+        // cannot be accounted for, these superfluous tokens will be burned,
+        // this is because there is no way to tell which participant (if any)
+        // had ownership over the token.
+        total_deposit = participant1_state.deposit + participant2_state.deposit;
+
+        participant1_amount = (
```

Comment:
This changed the semantics of the code, the order of payments is not arbitrary, the node that **did not** close the channel must be paid first.

Reply:
Why is that? I don't really get it.